### PR TITLE
Charlesmchen/group offers

### PR DESCRIPTION
**Pull request by charlesmchen**
This PR is now ready for review.

screen shot 2017-08-21 at 5 04 43 pm

PTAL @michaelkirk

### CODE REVIEW

**Comment by charlesmchen** [comment]
This PR is now ready for review.

PTAL @michaelkirk

File: Signal/src/UIColor+OWS.h [hunk 1]
```diff
@@ -14,6 +14,7 @@
 + (UIColor *)ows_darkGrayColor;
 + (UIColor *)ows_yellowColor;
 + (UIColor *)ows_reminderYellowColor;
++ (UIColor *)ows_reminderDarkYellowColor;
```
Comment (charlesmchen) [inline Signal/src/UIColor+OWS.h]: See the screenshot.  White text doesn't show up well against the "reminder yellow" so I used a similar but darker yellow that WhatsApp uses for these kinds of purposes.

File: Signal/src/ViewControllers/AppSettingsViewController.m [hunk 2]
```diff
@@ -354,9 +354,7 @@ - (void)unregisterUser
                                                       handler:^(UIAlertAction *action) {
                                                           [self proceedToUnregistration];
                                                       }]];
-    [alertController addAction:[UIAlertAction actionWithTitle:NSLocalizedString(@"TXT_CANCEL_TITLE", @"")
-                                                        style:UIAlertActionStyleCancel
-                                                      handler:nil]];
+    [alertController addAction:[OWSAlerts cancelAction]];
```
Comment (charlesmchen) [inline Signal/src/ViewControllers/AppSettingsViewController.m]: I DRY'd up the cancel actions.

File: Signal/src/util/ThreadUtil.m [hunk 3]
```diff
@@ -422,7 +422,7 @@ + (ThreadDynamicInteractions *)ensureDynamicInteractionsForThread:(TSThread *)th
             if (!hasUnwhitelistedMember) {
                 // Don't show offer in group thread if all members are already individually
                 // whitelisted.
-                hasUnwhitelistedMember = YES;
+                shouldHaveAddToProfileWhitelistOffer = NO;
```
Comment (charlesmchen) [inline Signal/src/util/ThreadUtil.m]: A small but important fix.

File: Signal/src/ViewControllers/AppSettingsViewController.m (hunk 2 above)
Comment (michaelkirk) [inline Signal/src/ViewControllers/AppSettingsViewController.m]: 👏 👏 👏

File: Signal/src/ViewControllers/ConversationView/MessagesViewController.m [hunk 4]
```diff
@@ -829,6 +838,30 @@ - (void)blockBannerViewWasTapped:(UIGestureRecognizer *)sender
...
+- (void)groupProfileWhitelistBannerWasTapped:(UIGestureRecognizer *)sender
+{
+    if (sender.state != UIGestureRecognizerStateRecognized) {
+        return;
+    }
+
+    UIAlertController *alertController =
+        [UIAlertController alertControllerWithTitle:nil message:nil preferredStyle:UIAlertControllerStyleActionSheet];
+
+    UIAlertAction *leaveAction = [UIAlertAction
```
Comment (michaelkirk) [inline Signal/src/ViewControllers/ConversationView/MessagesViewController.m]: `leaveAction` -> `whitelistAction`?

**Pull request merged**